pkg/kube/wait.go

```diff
@@ -88,6 +88,36 @@ func (c *Client) waitForResources(timeout time.Duration, created Result) error {
 					currentDeployment,
 				}
 				deployments = append(deployments, newDeployment)
+			case *appsv1beta1.Deployment:
+				currentDeployment, err := kcs.ExtensionsV1beta1().Deployments(value.Namespace).Get(value.Name, metav1.GetOptions{})
```

Comment:
Looking at this, I wonder if it shouldn't use the client corresponding to the API version.

`kcs.AppsV1beta1().Deployments`

If this is a problem, then it has already existed wit the old code.

Reply:
@unguiculus I thought the same thing and tried to test this change. The problem is that `deploymentutil.GetNewReplicaSet` (a package from within k8s), which is used shortly after this and is passed the `currentDeployment`, takes as an input something using the extensions v1beta1 interface. Making your suggested switch causes that code to break.

I expect there is a better way to clean this up but I'm not sure it needs to happen in this PR.